chain-abstraction/sol-implementation/edge_challenge_manager_test.go

```diff
@@ -260,6 +261,125 @@ func TestEdgeChallengeManager_Bisect(t *testing.T) {
 	})
 }
 
+func TestEdgeChallengeManager_AddSubchallengeLeaf(t *testing.T) {
```

Comment:
We didn't have a pure test that just checks if we can open a subchallenge. I added one and beefed it up to 5 challenge levels